server/server.go

```diff
@@ -731,6 +731,17 @@ func (s *Server) checkAuth(c *client) bool {
 	}
 }
 
+// Check that number of clients is below Max connection setting.
+func (s *Server) checkMaxConn(c *client) bool {
+	if c.typ == CLIENT {
```

Comment:
If we have added to s.clients, we could just do the following since if its a route will be ok I think.

s.mu.Lock()
defer s.mu.Unlock()
return len(s.clients) <+ s.opts.MaxConn